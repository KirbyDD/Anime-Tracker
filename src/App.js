import React, { Component } from 'react';
import Navbar from './Navbar';
import AnimeContainer from './AnimeContainer'
import './App.scss';

class App extends Component {
  constructor() {
    super();
    this.state = {
      animeList: []
    }
  }

  render() {
    return(
      <main>
        <h1>Working</h1>
        <Navbar />
        <AnimeContainer />
      </main>
    )
  }
}

export default App;
