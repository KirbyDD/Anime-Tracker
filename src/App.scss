.app{
  background: black;
  color: gold;
}

h1 {
  text-align: center;
}

