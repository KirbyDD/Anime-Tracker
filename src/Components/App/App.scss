.app {
  color: gold;
}

.background-photo {
  position: fixed;
  filter: brightness(20%);
  height: 100%;
  width: 100%;
  z-index: -1;
}

h1 {
  text-align: center;
  font-size: 30px;
  margin-top: 0px;
}

header {
  background-color: rgb(37, 37, 35);
  padding: 17px;
  z-index: 1;
}

button {
  height: 50px;
  width: 130px;
}
