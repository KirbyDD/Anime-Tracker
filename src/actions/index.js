export const addShow = newShow => ({
  type: 'ADD_SHOW',
  newShow
})

export const removeShow = id => ({
  type: 'REMOVE_SHOW',
  id
})

export const updateShow = id => ({
  type: 'UPDATE_SHOW',
  id
})

export const gatherAnime = data => ({
  type: 'GATHER_ANIME',
  data
})