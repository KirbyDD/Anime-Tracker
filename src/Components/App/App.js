import React, { Component } from 'react';
import {Route} from 'react-router-dom';
import {fetchAnime, fetchAdditionalAnime} from '../../apiCalls'
import {gatherAnime, gatherMoreAnime} from '../../actions/index';
import Navbar from '../Navbar/Navbar';
import AnimeContainer from '../AnimeContainer/AnimeContainer'
import dataCleaner from '../../dataCleaner';
import {connect} from 'react-redux';
import './App.scss';
import AnimeDetails from '../AnimeDetails/AnimeDetails';
import WatchListContainer from '../WatchListContainer/WatchListContainer';
import WatchListDetails from '../WatchListDetails/WatchListDetails';


class App extends Component {
  constructor() {
    super();
    this.state = {
      animeList: []
    }
  }
  async componentDidMount(){
    try{
      const result = await fetchAnime()
      const cleanData = dataCleaner(result.data)
      this.props.gatherAnime(cleanData)
      const result2 = await fetchAdditionalAnime()
      const cleanData2 = dataCleaner(result2.data)
      this.props.gatherMoreAnime(cleanData2)
    
    } catch(error) {
      console.log(error)
    }


  }

  addToList = newShow => {
    this.setState({ animeList: [...this.state.animeList, newShow]}) 
  }

  render() {
    return(
      <main className="app">
        <img src='https://images6.alphacoders.com/656/thumb-1920-656029.png' 
            alt='background with multiple anime characters'
            className='background-photo' />
        <header>
        <h1>Anime Tracker</h1>
        <Navbar />
        </header>
        <Route exact path='/'
          render={() => (
            <AnimeContainer />
          )} />
        <Route exact path='/watchlist'
        render={() => (
          <WatchListContainer />
        )} />
        <Route exact path='/anime/:id'
          render={({match}) => {
            const id = match.params;
            const show = this.props.anime.find(anime => parseInt(anime.id) === parseInt(id.id))
            return <AnimeDetails {...show} />
        }} />
        <Route exact path='/watchlist/:id'
          render={({match}) => {
            const id = match.params
            const show = this.props.followedShows.find(anime => parseInt(anime.id) === parseInt(id.id))
            console.log(show)
            return <WatchListDetails {...show} />
          }}
          />
      </main>
    )
  }
}

const mapStateToProps = store => ({
  ...store
})

const mapDispatchToProps = dispatch => ({
  gatherAnime: data => dispatch(gatherAnime(data)),
  gatherMoreAnime: data => dispatch(gatherMoreAnime(data))
})

export default connect(mapStateToProps, mapDispatchToProps)(App);
