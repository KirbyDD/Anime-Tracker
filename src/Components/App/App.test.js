import React from "react";
import { shallow } from "enzyme";
import { App, mapStateToProps, mapDispatchToProps } from "./App";
import { gatherAnime, gatherMoreAnime } from '../../actions/index'

let animeList = [
  {
    name: "anime1",
    id: 1
  },
  {
    name: "anime2",
    id: 2
  },
  {
    name: "anime3",
    id: 3
  }
];

let props = {
  id: 1,
  synopsis: "This is the synopsis",
  enTitle: "Life of TronKat",
  jpTitle: "Tronkat no Life desu",
  episodes: 27,
  posterImage: {
    large: "https://i.imgur.com/vihivug.jpg",
    tiny: null
  },
  coverImage: {
    small: "https://i.imgur.com/vihivug.jpg",
    tiny: null
  },
  rating: "99.9",
  startDate: "1990-01-26",
  endDate: "Still Playing",
  ageRating: "17+"
};

let followedShows = [
  {
    id: 1,
    synopsis: "This is the synopsis",
    enTitle: "Life of TronKat",
    jpTitle: "Tronkat no Life desu",
    episodes: 27,
    posterImage: {
      large: "https://i.imgur.com/vihivug.jpg",
      tiny: null
    },
    coverImage: {
      small: "https://i.imgur.com/vihivug.jpg",
      tiny: null
    },
    rating: "99.9",
    startDate: "1990-01-26",
    endDate: "Still Playing",
    ageRating: "17+"
  }
];

describe("App", () => {
  let wrapper;

  beforeEach(function() {
    wrapper = shallow(<App />);

    window.fetch = jest.fn().mockImplementation(() =>
      Promise.resolve({
        json: () => Promise.resolve(animeList)
      })
    );
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
})

  describe('mapStateToProps', () => {
    it('should return object with anime array', () => {
      let mockState = {
        anime: props,
        followedShows
      }
    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps).toEqual(mockState)
  })

  describe('mapDispatchToProps', () => {
    it('calls dispatch with an addShow action when addToWatchList is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = gatherAnime([props]);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.gatherAnime([props]);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('calls dispatch with an removeShow action when removeFromWatchList is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = gatherMoreAnime([props]);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.gatherMoreAnime([props]);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  })
});
