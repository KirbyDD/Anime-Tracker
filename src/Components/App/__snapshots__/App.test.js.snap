// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<main
  className="app"
>
  <img
    alt="background with multiple anime characters"
    className="background-photo"
    src="https://images6.alphacoders.com/656/thumb-1920-656029.png"
  />
  <header>
    <h1>
      Anime Tracker
    </h1>
    <Navbar />
  </header>
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/watchlist"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/anime/:id"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/watchlist/:id"
      render={[Function]}
    />
    <Route
      render={[Function]}
    />
  </Switch>
</main>
`;
